lib/src/main/kotlin/com/lemonappdev/konsist/api/ext/list/KoGenericTypeProviderListExt.kt

```diff
@@ -0,0 +1,83 @@
+@file:Suppress("detekt.TooManyFunctions")
+
+package com.lemonappdev.konsist.api.ext.list
+
+import com.lemonappdev.konsist.api.declaration.type.KoBaseTypeDeclaration
+import com.lemonappdev.konsist.api.provider.KoGenericTypeProvider
+import kotlin.reflect.KClass
+
+/**
+ * Returns a list containing generic type declarations.
+ */
+val <T : KoGenericTypeProvider> List<T>.genericTypes: List<KoBaseTypeDeclaration>
+    get() = mapNotNull { it.genericType }
+
+/**
+ * Filters the list to include only elements with a generic type satisfying the given predicate.
+ *
+ * @param predicate A function that defines the condition to be met by the generic type.
+ * @return A list containing elements with a generic type satisfying the predicate.
```

Comment:
In general different types of "@ elements" should be separated by a new line

```
* @param predicate A function that defines the condition to be met by the generic type.
* 
* @return A list containing elements with a generic type satisfying the predicate.
```

Reply:
Will be updated in this PR: https://lemonappdev.atlassian.net/browse/KON-660